Core/src/org/sleuthkit/autopsy/contentviewers/SQLiteViewer.java

```diff
@@ -363,33 +410,103 @@ protected Integer doInBackground() throws Exception {
+                        readTable(tableName, (currPage - 1) * ROWS_PER_PAGE + 1, ROWS_PER_PAGE);
+                    } else {
+                        nextPageButton.setEnabled(false);
+                        selectedTableView.setupTable(Collections.emptyList());
+                    }
+
                 } catch (InterruptedException | ExecutionException ex) {
                     LOGGER.log(Level.SEVERE, "Unexpected exception while reading table.", ex); //NON-NLS
                 }
             }
-        }.execute();
-
+        };
+        worker.execute();
     }
 
-    enum SQLStorageClass {
-        NULL,
-        INTEGER,
-        REAL,
-        TEXT,
-        BLOB
-    };
+    private void readTable(String tableName, int startRow, int numRowsToRead) {
 
-    private class SQLColDef {
+        if (worker != null && !worker.isDone()) {
+            worker.cancel(false);
+            worker = null;
+        }
 
-        private final String colName;
-        private final SQLStorageClass storageClass;
+        worker = new SwingWorker<ArrayList<Map<String, Object>>, Void>() {
+            @Override
+            protected ArrayList<Map<String, Object>> doInBackground() throws Exception {
+                try {
+                    Statement statement = connection.createStatement();
+                    ResultSet resultSet = statement.executeQuery(
+                            "SELECT * FROM " + tableName
+                            + " LIMIT " + Integer.toString(numRowsToRead)
+                            + " OFFSET " + Integer.toString(startRow - 1)
+                    ); //NON-NLS
 
-        SQLColDef(String colName, SQLStorageClass sc) {
-            this.colName = colName;
-            this.storageClass = sc;
+                    return resultSetToArrayList(resultSet);
+                } catch (SQLException ex) {
+                    LOGGER.log(Level.SEVERE, "Failed to get data for table.", ex); //NON-NLS
+                }
+                //NON-NLS
+                return null;
+            }
+
+            @Override
+            protected void done() {
+
+                if (isCancelled()) {
+                    return;
+                }
+
+                super.done();
+                try {
+                    ArrayList<Map<String, Object>> rows = get();
+                    if (Objects.nonNull(rows)) {
+                        selectedTableView.setupTable(rows);
+                    }else{
+                        selectedTableView.setupTable(Collections.emptyList());
+                    }
+                } catch (InterruptedException | ExecutionException ex) {
+                    LOGGER.log(Level.SEVERE, "Unexpected exception while reading table.", ex); //NON-NLS
+                }
+            }
+        };
+
+        worker.execute();
+    }
+
+    private ArrayList<Map<String, Object>> resultSetToArrayList(ResultSet rs) throws SQLException {
+        ResultSetMetaData md = rs.getMetaData();
```

Comment:
This would be a bit easier to read and more in line with our coding guidelines if we used a longer variable name than "md." I personally experienced having to look back at the variable definition to remind myself what "md" stands for while reading the code, short as it is. The use of "i" as an index and "rs" for a record set is so common the short names work, but I don't think "md" is a common abbreviation.